VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -25,17 +25,924 @@ if ($PSEdition -eq 'Desktop') {
+}
+Export-ModuleMember -Function Confirm-ESXiCertificateAlreadyInstalled
+
+Function Confirm-CAInVcenterServer {
+    <#
+    Verify the root certificate thumbprint matches with one of the CA thumbprints from vCenter server
+
+    .DESCRIPTION
+    This cmdlet will get the thumbprint from the root certificate and matches it with the CA thumbprint from vCenter. 
+    You need to pass in the complete path for the certificate file. 
+    Returns true if thumbprint matches, else returns false.
+
+
+    .EXAMPLE
+    Confirm-CAInVcenterServer -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re123! -domain sfo-m01 -issuer rainpole -signedCertificate F:\powershell-module-for-vmware-cloud-foundation-certificate-management\Root64.cer
+    This command will match the thumbprint of provided root certificate file with the thumbprints on the vCenter server matching the issuer "rainpole"
+    #>
+    
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $server,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $pass,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $domain,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $signedCertificate,
+        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [String] $issuer
+    )
+    
+    try {
+
+        if ($PsBoundParameters.ContainsKey("issuer")) { 
+            $vcThumbprints = Get-vCenterCertificateThumbprints -server $server -user $user -pass $pass -domain $domain -issuer $issuer
+        }
+        else {
+            $vcThumbprints = Get-vCenterCertificateThumbprints -server $server -user $user -pass $pass -domain $domain
+        }
+        if (Test-Path $signedCertificate -PathType Leaf ) {
+            Write-Host "Certificate file found - $signedCertificate"
+        }
+        else {
+            Write-Error "Could not find certificate in $signedCertificate." -ErrorAction Stop
+            return
+        }
+        $crt = New-Object System.Security.Cryptography.X509Certificates.X509Certificate
+        $crt.Import($signedCertificate)
+        $signedCertThumbprint = $crt.GetCertHashString()
+
+        $match = $false
+        Foreach ($vcThumbprint in $vcThumbprints) { 
+            if ($vcThumbprint -eq $signedCertThumbprint) {
+                Write-Host "Signed certificate thumbprint matches with the vCenter server CA thumbprint"
+                $match = $true
+                break
+            }
+        }
+        if (!$match) {
+            Write-Error "Signed certificate thumbprint DOESNT match with any of the vCenter server certificates thumbprints"
+        }
+        return $match    
+    }
+    Catch {
+        Debug-ExceptionWriter -object $_
+    }
+}
+
+Export-ModuleMember -Function Confirm-SignedCertificateWithCA
+
+Function Get-EsxiCSR {
```

Comment:
Should this be `Request-EsxiCsr`?

Reply:
CSR itself as Certificate signed request 
Request-EsxiCsr , do you think it is better?

Reply:
Changed to Request-EsxiCsr. Makes more sense now.